src/config.hpp.in

```diff
@@ -47,5 +47,6 @@
 #cmakedefine CHAI_ENABLE_IMPLICIT_CONVERSIONS
 #cmakedefine CHAI_DISABLE_RM
 #cmakedefine CHAI_ENABLE_UM
+#cmakedefine CHAI_ARRAY_BOUNDS_CHECK
```

Comment:
This should be `CHAI_ENABLE_*`, matching the option from the CMakeLists, prefixed with `CHAI_`